modules/navier_stokes/src/linearfvkernels/LinearFVTurbulentLimitedAdvection.C

```diff
@@ -0,0 +1,179 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LinearFVTurbulentLimitedAdvection.h"
+#include "MooseLinearVariableFV.h"
+#include "NSFVUtils.h"
+#include "NavierStokesMethods.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", LinearFVTurbulentLimitedAdvection);
+
+InputParameters
+LinearFVTurbulentLimitedAdvection::validParams()
+{
+  InputParameters params = LinearFVScalarAdvection::validParams();
+  params.addClassDescription("Represents the matrix and right hand side contributions of an "
+                             "advection term for a turbulent variable.");
+
+  params.addParam<std::vector<BoundaryName>>(
+      "walls", {}, "Boundaries that correspond to solid walls.");
+
+  return params;
+}
+
+LinearFVTurbulentLimitedAdvection::LinearFVTurbulentLimitedAdvection(const InputParameters & params)
+  : LinearFVScalarAdvection(params),
+    _advected_interp_coeffs(std::make_pair<Real, Real>(0, 0)),
+    _wall_boundary_names(getParam<std::vector<BoundaryName>>("walls"))
+{
+  Moose::FV::setInterpolationMethod(*this, _advected_interp_method, "advected_interp_method");
+}
+
+void
+LinearFVTurbulentLimitedAdvection::initialSetup()
+{
+  LinearFVScalarAdvection::initialSetup();
+  NS::getWallBoundedElements(
+      _wall_boundary_names, _fe_problem, _subproblem, blockIDs(), _wall_bounded);
+}
+
+void
+LinearFVTurbulentLimitedAdvection::addMatrixContribution()
+{
+  // Coumputing bounding map
+  const Elem * elem = _current_face_info->elemPtr();
+  const auto bounded_elem = _wall_bounded.find(elem) != _wall_bounded.end();
+  const Elem * neighbor = _current_face_info->neighborPtr();
+  const auto bounded_neigh = _wall_bounded.find(neighbor) != _wall_bounded.end();
+
+  // If we are on an internal face, we populate the four entries in the system matrix
+  // which touch the face
+  if (_current_face_type == FaceInfo::VarFaceNeighbors::BOTH)
```

Comment:
let's discuss this tomorrow. this is a ton code duplication for something that block restriction should achieve naturally?

Reply:
if you dont think so, then let's create "addMatrixContrib/ETCInternal in the advection class.

then in the base class (regular advection) you do: addMatrixContrib() { addMatrixContribInternal() }

and in the derived class, you do 
if (on boundary)
  code code
else
 addMatrixContribInternal();

Reply:
I am tabling this for now since this is a bit unrelated to this PR. Let's keep this optimization for future efforts.